MoltenVK/MoltenVK/GPUObjects/MVKPipeline.h

```diff
@@ -323,6 +327,8 @@ class MVKGraphicsPipeline : public MVKPipeline {
 
 	MVKSmallVector<VkViewport, kMVKCachedViewportScissorCount> _viewports;
 	MVKSmallVector<VkRect2D, kMVKCachedViewportScissorCount> _scissors;
+	MVKSmallVector<VkDynamicState> _dynamicState;
```

Comment:
I wonder if this wouldn't be better suited to a set-like data structure (e.g. `std::set`).

Reply:
Yeah. I flipped a coin on this one. The number of dynamic states on any pipeline is so small I figured the retrieval overhead would be small compared to the memory and insertion overhead of a set.